### PR TITLE
Create placeholder pages for Student table

### LINKED ISSUE #45: Subfeature: Create placeholder pages for index, create, edit for Students table
Part of Epic #6

* [ ] Create a placeholder index page that you can navigate to from the admin dropdown menu in the navigation bar
* [ ] Create a create student page that you can navigate to from the index page
* [ ] Create an edit page that you can navigate to from the index page
* [ ] Create test files for each page (w/ placeholder tests)

### PR DESCRIPTION
**Pull request by justinwu13**
## Overview

In this PR, we create placeholder pages for the Student table. The index page can be accessed via the Admin dropdown menu in the navigation bar, and the index page contains links to the create and edit pages. Pages all contain placeholder text for now.

## Screenshots (Optional)

Screenshot 2024-11-21 at 5 49 46 PM
Screenshot 2024-11-21 at 5 50 02 PM

## Feedback Request (Optional)

Please give suggestions on the following:
- The naming scheme is a little different here than in our previous assignments. There's a little bit of inconsistency with the URLs and the names of the pages, could be changed later if this is an issue.

## Future Possibilities (Optional)

Some possible points could be:
- This is part of epic #6. These pages will function similarly to the existing Commons pages, where the index page contains a table of Students, the create page will contain the form to create a new Student, and the edit page will allow you to alter existing Students.

## Validation (Optional)

1. Download this branch.
2. Run the project.
3. (Admins only) Navigate to the Admin dropdown in the NavBar, and click on Students.
4. Navigate to the other pages via the links on the Student index page.

Alternatively:
1. Use my dokku deploy: https://happycows-dev2-justinwu13.dokku-10.cs.ucsb.edu/
2. (Admins only) Navigate to the Admin dropdown in the NavBar, and click on Students.
3. Navigate to the other pages via the links on the Student index page.

## Tests

- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues

Closes #45

### CODE REVIEW

File: frontend/src/main/pages/AdminStudentsCreatePage.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import BasicLayout from "main/layouts/BasicLayout/BasicLayout";
+
+export default function AdminStudentsCreatePage() {
+  // Stryker disable all : placeholder for future implementation
```
Comment (r0ckwav3) [inline frontend/src/main/pages/AdminStudentsCreatePage.js:4]: Are these stryker disables necessary? I think your tests should cover mutations here

**Review by r0ckwav3 — CHANGES_REQUESTED** [review]
remove strykers if they're no longer necessary

File: frontend/src/main/pages/AdminStudentsCreatePage.js (hunk 1 above)
Comment (r0ckwav3) [reply frontend/src/main/pages/AdminStudentsCreatePage.js:4]: just tested it locally, they should be safe to remove

Comment (justinwu13) [reply frontend/src/main/pages/AdminStudentsCreatePage.js]: removed them and verified that mutation testing still passes

**Comment by justinwu13** [comment]
thanks for catching that, removed and verified that it still passes testing on my local

**Review by r0ckwav3 — APPROVED** [review]
Everything looks good now!

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**